subsystem/cluster_v2_test.go

```diff
@@ -0,0 +1,118 @@
+package subsystem
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/go-openapi/swag"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/assisted-service/client/installer"
+	"github.com/openshift/assisted-service/internal/host/hostutil"
+	"github.com/openshift/assisted-service/models"
+)
+
+var _ = Describe("[V2ClusterTests]", func() {
+	ctx := context.Background()
+	var cluster *models.Cluster
+	var clusterID strfmt.UUID
+	var infraEnv *models.InfraEnv
+	var infraEnvID strfmt.UUID
+
+	AfterEach(func() {
+		clearDB()
+	})
+
+	BeforeEach(func() {
+		clusterReq, err := userBMClient.Installer.RegisterCluster(ctx, &installer.RegisterClusterParams{
+			NewClusterParams: &models.ClusterCreateParams{
+				Name:                     swag.String("test-cluster"),
+				OpenshiftVersion:         swag.String(openshiftVersion),
+				PullSecret:               swag.String(pullSecret),
+				BaseDNSDomain:            "example.com",
+				ClusterNetworkHostPrefix: 23,
+			},
+		})
+
+		Expect(err).NotTo(HaveOccurred())
+		cluster = clusterReq.GetPayload()
+		Expect(swag.StringValue(cluster.Status)).Should(Equal("insufficient"))
+
+		infraReq, err := userBMClient.Installer.RegisterInfraEnv(ctx, &installer.RegisterInfraEnvParams{
+			InfraenvCreateParams: &models.InfraEnvCreateParams{
+				Name:             swag.String("test-infra-env"),
+				OpenshiftVersion: swag.String(openshiftVersion),
+				PullSecret:       swag.String(pullSecret),
+				SSHAuthorizedKey: swag.String(sshPublicKey),
+				ImageType:        models.ImageTypeFullIso,
+			},
+		})
+
+		Expect(err).NotTo(HaveOccurred())
+		infraEnv = infraReq.GetPayload()
+	})
+
+	JustBeforeEach(func() {
```

Comment:
Why separate this out?
I don't really think this is the intended use of `JustBeforeEach`
ref: https://onsi.github.io/ginkgo/#separating-creation-and-configuration-justbeforeeach